File: export_leads.js
require('dotenv').config();
const { MongoClient } = require('mongodb');
const fs = require('fs');
const path = require('path');

const MONGODB_URI = process.env.MONGODB_URI;

async function exportLeads() {
  if (!MONGODB_URI) {
    console.error("MONGODB_URI is not set in .env file");
    return;
  }

  const client = new MongoClient(MONGODB_URI);

  try {
    console.log("Connecting to MongoDB...");
    await client.connect();
    console.log("Connected successfully.");

    const db = client.db(); 
    const crmCollection = db.collection('crm');

    const doc = await crmCollection.findOne({ _id: 'main' });

    if (!doc || !doc.leads || doc.leads.length === 0) {
      console.log("No leads found in the database.");
      return;
    }

    const leads = doc.leads;
    console.log(`Found ${leads.length} leads. Generating CSV...`);

    const headers = [
      'Name',
      'Company',
      'Mobile',
      'Email',
      'Requirement',
      'Location',
      'Date'
    ];

    let csvContent = headers.join(',') + '\n';

    leads.forEach(lead => {
      const clean = (str) => {
        if (!str) return '';
        let cleaned = String(str).replace(/"/g, '""').replace(/\r?\n|\r/g, ' ');
        if (cleaned.includes(',') || cleaned.includes('"')) {
          return `"${cleaned}"`;
        }
        return cleaned;
      };

      const row = [
        clean(lead.SENDER_NAME || lead.sender_name),
        clean(lead.SENDER_COMPANY || lead.sender_company),
        clean(lead.SENDER_MOBILE || lead.sender_mobile),
        clean(lead.SENDER_EMAIL || lead.sender_email),
        clean(lead.SUBJECT || lead.subject || lead.product || lead.message),
        clean(lead.SENDER_CITY || lead.sender_city || lead.SENDER_STATE),
        clean(lead.QUERY_TIME || lead.query_time)
      ];

      csvContent += row.join(',') + '\n';
    });

    const filePath = path.join(__dirname, 'leads_backup.csv');
    fs.writeFileSync(filePath, csvContent, 'utf8');

    console.log(`\nSUCCESS! 🎉`);
    console.log(`Your leads have been saved to: ${filePath}`);
    console.log(`You can now open 'leads_backup.csv' with Microsoft Excel.`);

  } catch (error) {
    console.error("Error exporting leads:", error);
  } finally {
    await client.close();
  }
}

exportLeads();

File: scripts/export_leads.py
import os
import csv
from pathlib import Path
from dotenv import load_dotenv

load_dotenv()

BASE_DIR = Path(__file__).resolve().parent.parent
MONGODB_URI = os.getenv("MONGODB_URI")

def export_leads():
    if not MONGODB_URI:
        print("MONGODB_URI is not set in .env file")
        return

    try:
        from pymongo import MongoClient
        print("Connecting to MongoDB...")
        client = MongoClient(MONGODB_URI, serverSelectionTimeoutMS=5000)
        db = client.get_default_database()
        doc = db.crm.find_one({"_id": "main"})

        if not doc or "leads" not in doc or not doc["leads"]:
            print("No leads found in the database.")
            return

        leads = doc["leads"]
        print(f"Found {len(leads)} leads. Generating CSV...")

        file_path = BASE_DIR / "leads_backup.csv"
        headers = ["Name", "Company", "Mobile", "Email", "Requirement", "Location", "Date"]

        with open(file_path, "w", newline="", encoding="utf-8") as f:
            writer = csv.writer(f)
            writer.writerow(headers)
            for lead in leads:
                writer.writerow([
                    lead.get("SENDER_NAME") or lead.get("sender_name") or lead.get("name", ""),
                    lead.get("SENDER_COMPANY") or lead.get("sender_company") or lead.get("company", ""),
                    lead.get("SENDER_MOBILE") or lead.get("sender_mobile") or lead.get("phone", ""),
                    lead.get("SENDER_EMAIL") or lead.get("sender_email") or lead.get("email", ""),
                    lead.get("SUBJECT") or lead.get("subject") or lead.get("product") or lead.get("message", ""),
                    lead.get("SENDER_CITY") or lead.get("sender_city") or lead.get("city", ""),
                    lead.get("QUERY_TIME") or lead.get("query_time") or lead.get("createdAt", "")
                ])

        print(f"\nSUCCESS! 🎉")
        print(f"Your leads have been saved to: {file_path}")
        client.close()
    except Exception as e:
        print(f"Error exporting leads: {e}")

if __name__ == "__main__":
    export_leads()
